tensorflow_addons/rnn/cell.py

```diff
@@ -799,3 +799,50 @@ def get_config(self):
         }
         base_config = super().get_config()
         return {**base_config, **config}
+
+
+@tf.keras.utils.register_keras_serializable(package="Addons")
+class ZoneoutLSTMCell(keras.layers.LSTMCell):
+    """LSTM cell with recurrent zoneout.
+
+    https://arxiv.org/abs/1606.01305
+    """
+
+    @typechecked
+    def __init__(
+        self,
+        units: int,
+        zoneout_h: float = 0,
+        zoneout_c: float = 0,
+        seed: int = None,
+        **kwargs
+    ):
+        """
+        """
+        super().__init__(units, **kwargs)
+        self.zoneout_h = zoneout_h
+        self.zoneout_c = zoneout_c
+        self.seed = seed
+
+    def _zoneout(self, t, tm1, rate, training):
+        dt = tf.cast(
+            tf.random.uniform(t.shape, seed=self.seed) >= rate * training, t.dtype
```

Comment:
Instead of multiplying by `training`, it could be surrounded by and if/else statement so that we avoid unnecessary computation (the random matrix)

Reply:
If I understand things correctly, 99% of the execution time is a training. No harm at all but looks much nicer.

Reply:
I totally agree with you that it looks much nicer with the multiplications. But I am concerned that it could harm prediction performance. Sometimes we forget about the importance of predicting time as most of the trained models don't end in productions, but the main goal of training models is to use it in productions and therefore prediction time matters.

Reply:
Performance of production code is a much more broader topic. If someone report that that redundant random dramatically degrades their model inference to an unacceptable level, I will definitely try to find the best possible solution. Otherwise, I see no reasons to change it now and break something accidentally.
Also, just for note, branching is a bad thing for performance. Especially in the GPU world.

Reply:
It's ok, we can leave it in this more elegant way and take into account if someone reports any performance issue.

Reply:
Please note that training can be a tensor, which means if check will yield error. In the case you want to disable the zone out in the inference mode, please use tf.cond. Please check the keras.backend.dropout as an example.

Reply:
`tf.cond` is phasing out now: https://github.com/tensorflow/addons/issues/1794